examples/http_async_server.hpp

```diff
@@ -67,13 +71,138 @@ class http_async_server
             t.join();
     }
 
+    void
+    log_args()
+    {
+    }
+
+    template<class Arg, class... Args>
+    void
+    log_args(Arg const& arg, Args const&... args)
```

Comment:
Double checking that you really want `log_args()` to be `public`.  It side-steps the thread safety that you so nicely put into `log()`.  Your call.